lib/Driver/DarwinToolChains.cpp

```diff
@@ -284,6 +284,8 @@ toolchains::Darwin::constructInvocation(const LinkJobAction &job,
   case LinkKind::DynamicLibrary:
     Arguments.push_back("-dylib");
     break;
+  case LinkKind::StaticLibrary:
+    llvm_unreachable("invalid link kind");
```

Comment:
Rather than "invalid link kind" can you clarify that cannot perform archiving with the dynamic linker?